### PR TITLE
Add model graphbean

### PR DESCRIPTION
**Pull request by boersmamarcel**
**Dataset Addition Merge Checklist**

**Please ensure the following are completed before merging this pull request:**

- [x] Contributor signed the license agreement: To contribute to our repository, the contributor must sign the license agreement
- [x] Thorough Description: The contribution is adequately described, including origin, content, size, licensing, and the relevant academic paper.
- [x] Clear notebook guide on how to use the script.

Thanks for contributing

----
📚 Documentation preview 📚: https://fintorch--45.org.readthedocs.build/en/45/

### CODE REVIEW

File: fintorch/cli.py [hunk 1]
```diff
@@ -30,7 +34,49 @@ def train(model):
     """Train financial models"""
     # Implement your model training logic here
     click.echo(f"Training model: {model}")
+    cli = LightningCLI(
```
Comment (boersmamarcel) [inline fintorch/cli.py:37]: Skip this, and make this a sweep command for a particular model that is trained on a dataset.

File: fintorch/datasets/ellipticpp.py [hunk 2]
```diff
@@ -351,3 +359,173 @@ def process(self):
...
+            disjoint_train_ratio=0.3,
+            neg_sampling_ratio=2.0,
+            edge_types=[
+                ("wallets", "to", "transactions"),
+                ("transactions", "to", "wallets"),
+                ("wallets", "to", "wallets"),
+                ("transactions", "to", "transactions"),
+            ],
+        )
+        self.train_data, self.val_data, self.test_data = transform(dataset)
```
Comment (boersmamarcel) [inline fintorch/datasets/ellipticpp.py:465]: Transform per time slice and combine into train, validate, and test.

Comment (boersmamarcel) [reply fintorch/datasets/ellipticpp.py:465]: We have a set of graphs for each time step. Therefore, similar to https://github.com/pyg-team/pytorch_geometric/pull/7211 we might be able to solve it.

File: fintorch/models/graph/graphbean/graphBEAN.py [hunk 3]
```diff
@@ -0,0 +1,653 @@
...
+        hidden_channels (int): Number of hidden channels.
+        features_channels (dict): Dictionary specifying the number of channels for each edge type.
+
+    Attributes:
+        encoder_layers (nn.ModuleList): List of encoder layers.
+        decoder_layers (nn.ModuleList): List of feature decoder layers.
+        graph_edge_prediction (LinkClassifier): Link classifier for graph edge prediction.
+    """
+
+    def __init__(
```
Comment (boersmamarcel) [inline fintorch/models/graph/graphbean/graphBEAN.py:171]: Add dropout layer. Check edge message passing (add self connection).

File: fintorch/models/graph/graphbean/graphBEAN.py [hunk 4]
```diff
@@ -0,0 +1,653 @@
...
+        assert hidden_channels > 0, "Number of hidden channels must be greater than 0."
+
+        # Encoder layers
+        self.encoder_layers = nn.ModuleList()
+        first_encoder_layer = nng.HeteroConv(
+            {
+                edge_type: conv_type(-1, hidden_channels)
+                for edge_type in edge_types
+            },
+            aggr="sum",
```
Comment (boersmamarcel) [inline fintorch/models/graph/graphbean/graphBEAN.py:196]: Make this an argument

File: fintorch/models/graph/graphbean/wand/sweep.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
+import argparse
+import os
+
+import lightning as L
```
Comment (boersmamarcel) [inline fintorch/models/graph/graphbean/wand/sweep.py:4]: Clean up

File: fintorch/sweeps/wand/graphbean.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
+import argparse
+
+from lightning.pytorch.callbacks import ModelCheckpoint
+from lightning.pytorch.callbacks.early_stopping import EarlyStopping
+from lightning.pytorch.cli import LightningCLI
+
```
Comment (boersmamarcel) [inline fintorch/sweeps/wand/graphbean.py:6]: Remove file

File: setup.py [hunk 7]
```diff
@@ -10,16 +10,9 @@
...
-    "torch",
-    "torch_geometric",
-    "pytorch_lightning",
-    "kaggle",
-    "polars",
-    "numpy",
-    "huggingface_hub",
-    "seaborn",
-    "networkx",
+    "Click>=7.0", "torch", "torch_geometric", "lightning", "kaggle", "polars",
```
Comment (boersmamarcel) [inline setup.py:13]: Check clean install

File: tests/datasets/test_ellipticpp.py [hunk 8]
```diff
@@ -87,8 +89,62 @@ def test_process():
...
+    val_dataloader = data_module.val_dataloader()
+    assert val_dataloader is not None
+
+
+def test_link_neighbor_loader_type():
+    data_module = EllipticppDataModule(edge=("wallets", "to", "transactions"))
+    data_module.setup()
+    train_dataloader = data_module.train_dataloader()
+    assert isinstance(train_dataloader, LinkNeighborLoader)
+
```
Comment (boersmamarcel) [inline tests/datasets/test_ellipticpp.py:133]: Add more tests

File: fintorch/models/graph/graphbean/GraphBEANModule.yaml [hunk 9]
```diff
@@ -0,0 +1,14 @@
+model:
+  class_path: fintorch.models.graph.graphbean.graphBEAN.GraphBEANModule
+  init_args:
+    edge_types: ['transactions_to_wallets', 'wallets_to_wallets']
+trainer:
```
Comment (boersmamarcel) [inline fintorch/models/graph/graphbean/GraphBEANModule.yaml:5]: Check whether this runs!

**Comment by boersmamarcel** [comment]
Check list:

- [ ] Tutorials working
- [ ] Test coverage high enough
- [ ] Documentation added
- [ ] Run LightningCLI
- [ ] Run torch sweep model
